docs/intentReaderHandlers/RegisterProviderHandler.md

```diff
@@ -0,0 +1,187 @@
+# RegisterProviderHandler 
+
+## Overview
+
+This handler serves the purpose of simulating user inputs in FCA. Specific device calls prompt the system to request user input. When utilized alongside test providers found in src/providers and the SetApiResponseHandler, you gain the ability to provide mock responses to the device's provider calls. This is useful for manipulating various provider APIs such as keyboard and profile calls. Test provider will manage any provider that has refui. 
+
+It performs the following actions:
+1. Parses the input message received to get the provider.
+2. Informs FCA to use testProviders for setting the responses.
+3. Formats and sends the response back to the IntentReader.
+
+## Usage
+
+#### Use case - Requirement is to return abc@gmail.com for keyboard.email api
+
+* Register for the Keyboard Test Provider by sending the below intent - src/providers/KeyboardUIProviderTest.js
+*  
+```json
+
+    {
+        "action": "search",
+        "data": {
+            "query": {
+                "task": "registerProviderHandler",
+                "params": {
+                    "provider": "keyboard",
+                    "params": []
+                },
+                "action": "NA",
+                "appType": "firebolt"
+            }
+        },
+        "context": {
+            "source": "device"
+        }
+    }
+
+```
+* Set the response for keyboard.email using the SetApiResponseHandler by sending the below intent
+*   
+```json
+
+    {
+        "action": "search",
+        "data": {
+            "query": {
+                "task": "setApiResponse",
+                "params": {
+                    "apiResponse": {
+                        "module": "keyboard",
+                        "methodName": "keyboard.email",
+                        "type": "method",
+                        "attributes": [
+                            {
+                                "ApiText": "abc@gmail.com",
+                                "isCancelled": false,
+                                "withUi": false,
+                                "result": "john@doe.com"
+                            }
+                        ]
+                    }
```

Comment:
In this intent, the attributes doesnt match 

{
                                "ApiText": "abc@gmail.com",
                                "isCancelled": false,
                                "withUi": false,
                                "result": "john@doe.com"
                            }

why are ApiText and result different values ? Is that expected ?

Reply:
Updated as it is expected to be "abc@gmail.com".